modules/wiki/wiki.py

```
@@ -259,12 +262,12 @@ async def _callback(msg: Bot.MessageSession):
                                 i_msg_lst.append(Plain(session.locale.t('wiki.message.invalid_section.select')))
                                 i_msg_lst.append(Plain(session.locale.t('message.reply.prompt')))
 
-                            async def _callback(msg: Bot.MessageSession, forum_data=forum_data, r=r):
-                                display = msg.as_display(text_only=True)
-                                if isint(display) and int(display) <= len(forum_data) - 1:
-                                    await query_pages(session, title=forum_data[display]['text'], start_wiki_api=r.info.api)
+                                async def _callback(msg: Bot.MessageSession):
+                                    display = msg.as_display(text_only=True)
+                                    if isint(display) and int(display) <= len(forum_data) - 1:
```

Ruff (0.8.2)

Function definition does not bind loop variable `forum_data`

(B023)